x-pack/plugins/maps/public/layers/solution_layers/observability/metric_select.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { ChangeEvent } from 'react';
+import { EuiFormRow, EuiSelect, EuiSelectOption } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { OBSERVABILITY_LAYER_TYPE } from './layer_select';
+
+export enum OBSERVABILITY_METRIC_TYPE {
+  TRANSACTION_DURATION = 'TRANSACTION_DURATION',
+  SLA_PERCENTAGE = 'SLA_PERCENTAGE',
+  COUNT = 'COUNT',
+  UNIQUE_COUNT = 'UNIQUE_COUNT',
+}
+
+export function getMetricOptionsForLayer(layer: OBSERVABILITY_LAYER_TYPE): EuiSelectOption[] {
+  if (layer === OBSERVABILITY_LAYER_TYPE.APM_RUM_PERFORMANCE) {
+    return APM_RUM_PERFORMANCE_METRIC_OPTIONS;
+  }
+
+  if (layer === OBSERVABILITY_LAYER_TYPE.APM_RUM_TRAFFIC) {
+    return APM_RUM_TRAFFIC_METRIC_OPTIONS;
+  }
+
+  return [];
+}
+
+const APM_RUM_PERFORMANCE_METRIC_OPTIONS = [
```

Comment:
Could you move these global declarations to the top of the module iso relying on hoisting? it'd be more consistent with constants in other modules as well.